src/swsssdk/dbconnector.py

```diff
@@ -26,165 +29,384 @@ def load_sonic_db_config(sonic_db_file_path=SONIC_DB_CONFIG_FILE):
+    @staticmethod
+    def get_separator(db_name, ns_instance):
+        if SonicDBConfig._sonic_db_config_init == False:
+            SonicDBConfig.load_sonic_db_config()
+        SonicDBConfig.db_name_validation(db_name, ns_instance)
+        return SonicDBConfig._sonic_db_config[ns_instance]["DATABASES"][db_name]["separator"]
+
+class SonicV2Connector(object):
+    def __init__(self, use_unix_socket_path=True, **kwargs):
+        self.db_intfs = {}
+        self.clients = {}
+        self.ns_name = 'default'
+ 
+        ns_instances = self.get_db_connection_instances()
+        for instance in ns_instances:
+            if self.check_instance_valid(instance):
+                self.db_intfs[instance] = DBInterface(**kwargs)
+
+                #set a database name as a constant value attribute.
+                for db_name in self.get_db_list(instance):
+                    setattr(self, db_name, db_name)
+            else:
+                self.db_intfs[instance] = None
         self.use_unix_socket_path = use_unix_socket_path
-        for db_name in self.get_db_list():
-            # set a database name as a constant value attribute.
-            setattr(self, db_name, db_name)
 
-    def connect(self, db_name, retry_on=True):
+    def db_connect(self, db_name, retry_on=True, namespace='default'):
+        if self.check_instance_valid(namespace) == False:
+            self.clients[namespace] = None
+            return
+
         if self.use_unix_socket_path:
-            self.redis_kwargs["unix_socket_path"] = self.get_db_socket(db_name)
-            self.redis_kwargs["host"] = None
-            self.redis_kwargs["port"] = None
+            self.db_intfs[namespace].redis_kwargs["unix_socket_path"] = self.get_db_socket(db_name, namespace)
+            self.db_intfs[namespace].redis_kwargs["host"] = None
+            self.db_intfs[namespace].redis_kwargs["port"] = None
+        else:
+            self.db_intfs[namespace].redis_kwargs["host"] = self.get_db_hostname(db_name, namespace)
+            self.db_intfs[namespace].redis_kwargs["port"] = self.get_db_port(db_name, namespace)
+            self.db_intfs[namespace].redis_kwargs["unix_socket_path"] = None
+
+        db_id = self.get_dbid(db_name, namespace)
+        self.db_intfs[namespace].connect(db_id, retry_on)
+        self.clients[namespace] = self.db_intfs[namespace].get_redis_client(db_id)
+
+    """
+    Connect to the DB instance specified by the namespace parameter in database_config.json file.
+    if namespace="default", it connects to the default instance of "db_name".
+    if namespace=<0...n>, any value between 0 and n,
+                          it connects to "db_name" instance in that namespace as per the INCLUDES.
+    if namespace="all", it connects to the global instance of "db_name" + namespace <0..n> as present in INCLUDES.
+    """
+    def connect(self, db_name, retry_on=True, namespace='default'):
+        self.ns_name = namespace
```

Comment:
>self.ns_name = namespace [](start = 8, length = 24)

we should throw NotImplmentedError when namespace is used with TCP port.

Reply:
Done, NotImplementedError is thrown if the user attempts to use TCP socket connection and to a db instance not in the 'default' redis instance.